docs/broker/filtering.md

```diff
@@ -0,0 +1,360 @@
+
+_This example assumes the repository name is available as a CloudEvents
+extension with name `repository`._
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    cel:
+      parseExtensions: true
+      expression: >
+        ce.type == "com.github.pull.create" ||
+         (ce.type == "com.github.issue.create" &&
+          ce.repository == "proposals")
+
+```
+
+### Parsed data
+
+Cannot be specified with the SourceAndType field.
+
+_The `>` syntax is a standard yaml multiline string._
+
+_Until [google/cel-go#203](https://github.com/google/cel-go/issues/203) is
+resolved, dynamic integer fields in CEL must be compared as floats._
+
+```yaml
+spec:
+  filter:
+    cel:
+      parseData: true
+      expression: >
+        ce.type == "dev.knative.observation" &&
+        data.latency > 300.0
+```
+
+## Caveats
+
+### CEL
+
+The main caveat is the choice of CEL as expression language.
+
+CEL is new and the language is not officially supported by Google. Participation
+by non-Google contributors is low. OPA is planning to replace their policy
+language Rego with CEL, but
+[no timeline has been announced for this migration](https://github.com/open-policy-agent/opa/issues/811#issuecomment-401844999).
+The KrakenD API Gateway has announced support for CEL
+[here](https://medium.com/devops-faith/krakend-api-gateway-0-9-released-9427c249dbcd).
+
+The CEL language is new and has limited exposure, despite being informed by
+experience and research at Google. We may discover issues that make it less
+suitable for Trigger filtering (e.g.
+[google/cel-go#203](https://github.com/google/cel-go/issues/203).
+
+If CEL turns out to be a liability, a new expression language could be added
+alongside CEL without breaking existing triggers, allowing for an orderly
+deprecation.
+
+For example, a javascript filter could be added without disrupting existing CEL
+filters:
+
+```yaml
+spec:
+  filter:
+    javascript:
+      expression: ce.type == "com.github.pull.create"
+```
+
+## Alternatives Considered
+
+The following alternative solutions were considered but rejected.
+
+### Javascript or Lua as expression language
+
+Trigger filtering is a very limited case for an embedded language. All that is
+needed from a language is the ability to write conditions that evaluate to true
+or false.
+
+Javascript and Lua are commonly used as embedded languages, but their
+capabilities extend well beyond the needs of Trigger filtering. Those
+capabilities come with complexity: Javascript and Lua programs may not provably
+terminate, may have memory leaks, and may consume excessive amounts of CPU. All
+these cases would need to be guarded against by any process involved in
+filtering. If the expression language is used for upstream filter propagation,
+this could be a large number of diverse processes.
+
+Due to this additional implementation complexity, both Javascript and Lua have
+been judged less suitable for Trigger filtering than CEL.
```

Comment:
For your consideration, CEL is also portable now as you can parse / check in Go and evaluate in either Go or C++ using the existing OSS libs. We'll consider supporting WASM and Java if evaluators are desired there as well.